backend/pkg/entities/state.go

```diff
@@ -0,0 +1,53 @@
+package entities
+
+import "fmt"
+
+var states = [...]string{
+	"rebooting",
+	"factory-reset",
+	"configuration-cloud",
+	"configuration-cloud-security",
+	"configuration-user",
+	"configuration-gateway",
+	"ready",
+}
+
+// State represents the application configuration status
+type State struct {
+	Type string
+}
+
+// StateValidationError represents a custom state validation error
+type StateValidationError struct{}
+
+// NewStateValidationError creates a new StateValidationError instance
+func NewStateValidationError() *StateValidationError {
+	return &StateValidationError{}
+}
+
+func (err *StateValidationError) Error() string {
+	return fmt.Sprintf("State isn't valid")
```

Comment:
Why use sprintf here if the string has no format string like `%s` or `%d` ?

Reply:
`return "State isn't valid"`

Reply:
Done.